shared/chat/conversation/container.js

```diff
@@ -2,169 +2,77 @@
+  startConversation: (users: Array<string>) => void,
 }
 
 const mapStateToProps = (state: TypedState, {routePath, routeState}) => {
   const selectedConversationIDKey = routePath.last()
-
-  const you = state.config.username || ''
-  const followingMap = state.config.following
-  const metaDataMap = state.chat.get('metaData')
-
-  if (Constants.isPendingConversationIDKey(selectedConversationIDKey)) {
-    const tlfName = Constants.pendingConversationIDKeyToTlfName(selectedConversationIDKey)
-    if (tlfName) {
-      const participants = List(tlfName.split(','))
-
-      return {
-        followingMap: pick(followingMap, participants.toArray()),
-        messages: List(),
-        metaDataMap: metaDataMap.filter((k, v) => participants.contains(v)),
-        moreToLoad: false,
-        participants,
-        rekeyInfo: null,
-        selectedConversationIDKey,
-        validated: true,
-        threadLoadedOffline: false,
-        you,
-      }
-    }
-  }
+  let finalizeInfo = null
+  let rekeyInfo = null
+  let supersedes = null
+  let supersededBy = null
+  let participants = List()
 
   if (selectedConversationIDKey !== Constants.nothingSelected) {
     const conversationState = state.chat.get('conversationStates').get(selectedConversationIDKey)
     if (conversationState) {
       const inbox = state.chat.get('inbox')
       const selected = inbox && inbox.find(inbox => inbox.get('conversationIDKey') === selectedConversationIDKey)
-      const participants = selected && selected.participants || List()
-      const rekeyInfo = state.chat.get('rekeyInfos').get(selectedConversationIDKey)
-
-      const supersedes = Constants.convSupersedesInfo(selectedConversationIDKey, state.chat)
-      const supersededBy = Constants.convSupersededByInfo(selectedConversationIDKey, state.chat)
-      const finalizeInfo = state.chat.get('finalizedState').get(selectedConversationIDKey)
-
-      return {
-        firstNewMessageID: conversationState.firstNewMessageID,
-        followingMap: pick(followingMap, participants.toArray()),
-        messages: conversationState.messages,
-        metaDataMap: metaDataMap.filter((k, v) => participants.contains(v)),
-        moreToLoad: conversationState.moreToLoad,
-        participants,
-        rekeyInfo,
-        selectedConversationIDKey,
-        validated: selected && selected.state === 'unboxed',
-        you,
-        supersedes,
-        supersededBy,
-        threadLoadedOffline: conversationState.loadedOffline,
-        finalizeInfo,
-        editingMessage: state.chat.get('editingMessage'),
-      }
+      participants = selected && selected.participants || List()
+      rekeyInfo = state.chat.get('rekeyInfos').get(selectedConversationIDKey)
+      finalizeInfo = state.chat.get('finalizedState').get(selectedConversationIDKey)
+      supersedes = Constants.convSupersedesInfo(selectedConversationIDKey, state.chat)
+      supersededBy = Constants.convSupersededByInfo(selectedConversationIDKey, state.chat)
```

Comment:
Hmm so this has to be done twice for 2 different containers? Maybe it's worth connecting this now if it's not too much work